phoenix-core-client/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -5052,6 +5070,39 @@ void ensureSystemTablesMigratedToSystemNamespace()
         }
     }
 
+    /**
+     * Acquire distributed mutex of sorts to make sure only one JVM is able to run the upgrade code by
+     * making use of HBase's checkAndPut api.
+     *
+     * @return true if client won the race, false otherwise
+     * @throws SQLException
+     */
+    @VisibleForTesting
+    public boolean acquireUpgradeBlockMutex()
+            throws SQLException {
+        try (Table sysMutexTable = getSysMutexTable()) {
+            final byte[] rowKey = Bytes.toBytes("BLOCK_UPGRADE");
```

Comment:
Do we need to do a checkAndPut ?

If we assume that the block is created and removed manually, then a simple read would suffice.
We already have a separate mechanism to make sure that only a single JVM is running the upgrade.